Import more data

xander-doom opened the pull request:
Added features:
- Category download (instead of TopViews) (specify categories in db/category.json)
- View threshold filtering (currently set to 300 views per day minimum)
- Prevent database entry duplication
- Added actual Gemini query
- "Enter" presses the "Generate theory" button
- "Generate theory" button is disabled during generation

Running the script may take up to an hour and will download 120mb of Wikipedia articles.

NOTE: Categories are a tree WITH RECURSION, so although the massviews website prevents duplicates, is is very possible for a theoretically small category (like "conspiracy theories") to include giant top level categories (like "history") which kind of makes the categories a poor filter. Oh well.

xander-doom committed:
Prevent duplicate imports

xander-doom committed:
Add download scripts for Massviews instead of Topviews

xander-doom committed:
Dockerize new scripts

xander-doom committed:
Slight prompt upgrade

xander-doom committed:
Disable "generate theory" button after pressing

xander-doom committed:
generate theory on "enter"

xander-doom committed:
Squashed commit of the following:

commit e42adecaa4da2391eafa139a38265faae49e562b
Merge: 59d3099 c9d1a27
Author: Xander Doom <43481280+xander-doom@users.noreply.github.com>
Date:   Tue Apr 8 17:57:31 2025 -0400

    Merge pull request #44 from ajvarchetti/UI_Improvements_Theme

    UI improvements theme

commit c9d1a27bf1998bbbd64a44aa82bf2eac823aa04a
Author: AJ Varchetti <varchettiaj@gmail.com>
Date:   Tue Mar 25 11:37:38 2025 -0400

    Removed commented test code

commit c02c6a814840004f98c97d8085872f26b35a1cfd
Author: AJ Varchetti <varchettiaj@gmail.com>
Date:   Tue Mar 25 11:36:09 2025 -0400

    Made website only light mode + added input sanitization.

commit 1bc52929b7d607b7e063d72ba9b76199ee09680b
Author: AJ Varchetti <varchettiaj@gmail.com>
Date:   Sun Mar 16 18:15:20 2025 -0400

    Improved layout and feel. Also further implemented new color scheme

xander-doom committed:
Merge branch 'main' into import-more-data

xander-doom committed:
Add views back to elasticsearch data

xander-doom committed:
Add views back to elasticsearch data

ajvarchetti commented:
The data download seems to work, but I get this error when I try to generate a theory. "2025-04-09 14:33:52 elasticsearch-wrapper-api-1  | ❌ Index 'wikipedia_conspiracies' does not exist
2025-04-09 14:33:52 elasticsearch-wrapper-api-1  | 172.18.0.1 - - [09/Apr/2025 18:33:52] "GET /generate?q=Marvel,NASA HTTP/1.1" 400 -"

xander-doom commented:
> The data download seems to work, but I get this error when I try to generate a theory. "2025-04-09 14:33:52 elasticsearch-wrapper-api-1 | ❌ Index 'wikipedia_conspiracies' does not exist 2025-04-09 14:33:52 elasticsearch-wrapper-api-1 | 172.18.0.1 - - [09/Apr/2025 18:33:52] "GET /generate?q=Marvel,NASA HTTP/1.1" 400 -"

Hmm it *shouldn’t* exist. I changed the index name to make more sense. Can you search your codebase for any references to “wikipedia_conspiracies”? You might need to delete your existing elasticsearch data or rebuild the api container.

ajvarchetti committed:
This fixes issue. Can confirm data loads now

ajvarchetti reviewed (approved):
This works on my machine, had to modify one file to get it to actually generate a theory. Good stuff

xander-doom committed:
Merge branch 'import-more-data' of github.com:ajvarchetti/Capstone_OSU into import-more-data

xander-doom committed:
Copy script files into Docker image

Had to switch the bind mounts because the files were being copied into a mounted directory, which meant they were being copied into the directory they were already in. Scripts on the server are now properly containerized (as long as they end with .py)
